### ISSUE TITLE
Bug: Opening Workspace dropdown does not put focus on input

### ISSUE THREAD

**Issue by nickytonline** [open-sauced/app#3362]
When you click or open the workspace dropdown via keyboard, the search input does not receive focus. This is annoying as a mouse user, but unusable as a keyboard user or some other form of assitive technology.

![CleanShot 2024-05-09 at 19 37 46](https://github.com/open-sauced/app/assets/833231/5ba28246-7034-406a-82fc-6ead9aafeb8d)

![CleanShot 2024-05-09 at 19 40 32](https://github.com/open-sauced/app/assets/833231/fa82d350-73cc-4401-a29c-9558dc76bb1d)

**Comment by nickytonline** [comment]
When the workspace dropdown is opened, it should give focus to the search input. This can be done with a `useEffect` and `useRef`.
